### PR TITLE
use Secure port for SSL given by TokenMapSupplier

### PR DESCRIPTION
**Pull request by shailesh33**
By default, secure port is the same as the dynomite listening port. This way, current OSS users using SSL will not break.
If the tokenMapSupplier provides a secure_port in the cluster_describe, that one will be used with SSL.

### CODE REVIEW

File: dyno-jedis/src/main/java/com/netflix/dyno/jedis/JedisConnectionFactory.java [hunk 1]
```diff
@@ -81,7 +81,7 @@ public JedisConnection(HostConnectionPool<Jedis> hostPool) {
 						hostPool.getSocketTimeout());
 			}
 			else {
-				jedisClient = new Jedis(host.getHostAddress(), host.getPort(), hostPool.getConnectionTimeout(),
+				jedisClient = new Jedis(host.getHostAddress(), host.getSecurePort(), hostPool.getConnectionTimeout(),
```
Comment (ipapapa) [inline dyno-jedis/src/main/java/com/netflix/dyno/jedis/JedisConnectionFactory.java]: Do we support secure port handling on Dynomite? Will this be half-baked if the server does not support it yet?

File: dyno-core/src/main/java/com/netflix/dyno/connectionpool/impl/lb/AbstractTokenMapSupplier.java [hunk 2]
```diff
@@ -227,13 +227,19 @@ private boolean isLocalDatacenterHost(Host host) {
                 String zone = (String) jItem.get("zone");
                 String datacenter = (String) jItem.get("dc");
                 String portStr = (String) jItem.get("port");
+                String securePortStr = (String) jItem.get("secure_port");
```
Comment (ipapapa) [inline dyno-core/src/main/java/com/netflix/dyno/connectionpool/impl/lb/AbstractTokenMapSupplier.java]: Does DM currently expose the secure port?

Comment (shailesh33) [reply dyno-core/src/main/java/com/netflix/dyno/connectionpool/impl/lb/AbstractTokenMapSupplier.java]: You reviewed it :)

File: dyno-jedis/src/main/java/com/netflix/dyno/jedis/JedisConnectionFactory.java (hunk 1 above)
Comment (shailesh33) [reply dyno-jedis/src/main/java/com/netflix/dyno/jedis/JedisConnectionFactory.java]: what do you mean by half-baked? It depends on how cluster_describe says.

**Pull request merged**